rules-reviewed/eap7/eap6/resteasy.windup.xml

```diff
@@ -0,0 +1,826 @@
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000007">
+            <when>
+                <javaclass references="org.jboss.resteasy.client.ClientRequestFactory" />
+            </when>
+            <perform>
+                <hint title="Deprecated ClientRequestFactory class" effort="1" severity="mandatory">
+                    <message>Replace the `org.jboss.resteasy.client.ClientRequestFactory` usage with `org.jboss.resteasy.client.jaxrs.org.jboss.resteasy.client.jaxrs.ResteasyClientBuilder` class.</message>
+                    <link title="RESTeasy Client Application Changes"
+                        href="https://access.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/7.0.beta/migration-guide/chapter-4-application-changes" />
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000008">
+            <when>
+                <javaclass references="org.jboss.resteasy.client.ClientResponseFailure" />
+            </when>
+            <perform>
+                <hint title="Deprecated ClientResponseFailure exception" effort="1" severity="mandatory">
+                    <message>Replace the `org.jboss.resteasy.client.ClientResponseFailure` with `javax.ws.rs.client.ResponseProcessingException` 
+                    or with `javax.ws.rs.client.ProcessingException` or with `javax.ws.rs.client.WebApplicationException` class.</message>
+                    <link title="RESTeasy Client Application Changes"
+                        href="https://access.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/7.0.beta/migration-guide/chapter-4-application-changes" />
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000009">
+            <when>
+                <javaclass references="org.jboss.resteasy.spi.StringConverter" />
+            </when>
+            <perform>
+                <hint title="RESTeasy SPI Application Change for StringConverter" effort="1" severity="mandatory">
+                    <message>Replace `org.jboss.resteasy.spi.StringConverter` by `javax.ws.rs.ext.ParamConverterProvider`.</message>
+                    <link title="RESTeasy SPI Application Changes"
+                        href="https://access.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/7.0.beta/migration-guide/chapter-4-application-changes" />
+                    <link title="JAX-RS API" href="https://jax-rs-spec.java.net/nonav/2.0-rev-a/apidocs/index.html?javax/ws/rs/ext/ParamConverterProvider.html" />
+                    <tag>resteasy</tag>
+                </hint>
+            </perform>
+        </rule>
+        <rule id="resteasy-eap6-000010">
```

Comment:
Similar rule should exist for `org.jboss.resteasy.spi.Registry`. They are mentioned together in the Migration Guide.